wayland-scanner/src/interfaces.rs

```diff
@@ -0,0 +1,106 @@
+use proc_macro2::{Ident, Span, TokenStream};
+
+use wayland_commons::scanner::{Interface, Message, Protocol, Type};
+
+use quote::quote;
+
+pub fn generate(protocol: &Protocol) -> TokenStream {
+    let interfaces = protocol.interfaces.iter().map(generate_interface);
+    quote!( #(#interfaces)* )
+}
+
+fn generate_interface(interface: &Interface) -> TokenStream {
+    let const_name = Ident::new(
+        &format!("{}_INTERFACE", interface.name.to_ascii_uppercase()),
+        Span::call_site(),
+    );
+    let iface_name = &interface.name;
+    let iface_version = interface.version;
+    let requests = build_messagedesc_list(&interface.requests, &interface.name);
+    let events = build_messagedesc_list(&interface.events, &interface.name);
+
+    quote!(
+        pub static #const_name: wayland_commons::Interface = wayland_commons::Interface {
```

Comment:
Usually in proc-macros you'd use absolute paths (`::wayland_commons::Interface`) but not sure whether that's important here.